tests/21-build_docker.sh

```diff
@@ -60,5 +60,55 @@ stest 1 sudo singularity build -F "$CONTAINER" docker://something_that_doesnt_ex
 stest 1 singularity exec "$CONTAINER" true
 stest 1 singularity exec "$CONTAINER" false
 
+# make sure local test does not exist, ignore errors
+sudo docker kill registry >/dev/null 2>&1
+sudo docker rm registry >/dev/null 2>&1
+
+# start local docker registry
+stest 0 sudo docker run -d -p 5000:5000 --restart=always --name registry registry:2
+# pull busybox from docker and push to local registry
+stest 0 sudo docker pull busybox
+stest 0 sudo docker tag busybox localhost:5000/my-busybox
+stest 0 sudo docker push localhost:5000/my-busybox
+
+# alright, now we have a local registry to test with
+
+# test with custom registry and custom namespace (including empty namespace)
+# from squashfs to squashfs (via def file)
+cat >"$DEFFILE" <<EOF
+Bootstrap: docker
+From: localhost:5000/my-busybox
+EOF
+
+stest 0 sudo SINGULARITY_NOHTTPS=true singularity build -F "$CONTAINER" "$DEFFILE"
+stest 0 singularity exec "$CONTAINER" true
+stest 1 singularity exec "$CONTAINER" false
+
+cat >"$DEFFILE" <<EOF
+Bootstrap: docker
+From: my-busybox
+Registry: localhost:5000
+EOF
+
```

Comment:
If the user exports an environment and deffile which takes preference? I can see arguments for both - an environment is more a run time thing and probably should take preference, but on the other hand, if there is an envar set (and the user forgot) it would produce confusing errors. What do you think?

Reply:
Not sure, and I'm not even sure which one singularity would pick right now.